# My notebooks

In [170]:
import pandas as pd

def nbs_to_df(nbs):
    parsed_notebooks = [parse(a_notebook.split()) for a_notebook in nbs]
    df = pd.DataFrame(parsed_notebooks, columns=['title', 'slug', 'total_votes', 'date'])
    df = df[df['total_votes'] > 0 ]
    df['url'] = df.apply(to_url, axis=1)
    df = df.sort_values("title", ascending=True)
    return df

def parse(a_notebook):
    slug = a_notebook[0]
    total_votes = int(a_notebook[-1])
    title = ' '.join(a_notebook[1:-6])
    date = ' '.join(a_notebook[-3:-2])
    return (title, slug, total_votes, date)

def to_url(row, include_bullet=True, include_votes=True):
    txt = f"[{row.title}](http://kaggle.com/{row.slug})"
    if include_bullet:
        txt = f"* {txt}"
    if include_votes:
        txt = f"{txt} ({row.total_votes})"
    return txt


def get_section(df, title, subtitle, url, matcher, sort_by):
    asc = sort_by != 'total_votes'
    if type(matcher) == str:
        dfx = df[df['title'].str.contains(matcher)]
    else:
        dfx = df[df.title.apply(matcher)]
    dfx = dfx.sort_values(sort_by, ascending=asc)
    
    txt = f"## [{title}]({url})\n"
    txt += f"**{subtitle}**\n\n"
    
    txt += '\n'.join(dfx['url'])
    return txt


In [171]:
nbs = !kaggle kernels list -m --page-size 100 --sort-by voteCount
df = nbs_to_df(nbs[2:])
len(df)

53

# By Competition

In [172]:
def is_other_competitions(title):
    for t in [ "🦠", "🇮🇳", "🐠", "📖", "H&M", "💲", "🐦", "☣️"]:
        if t in title:
            return False
    return True


COMPETITIONS = [
    ("Sartorius - Cell Instance Segmentation", 
     "Detect single neuronal cells in microscopy images", 
     "https://www.kaggle.com/c/sartorius-cell-instance-segmentation", 
     "🦠",
     'total_votes'),
    ("chaii - Hindi and Tamil Question Answering", 
     "Identify the answer to questions found in Indian language passages", 
     "https://www.kaggle.com/c/chaii-hindi-and-tamil-question-answering", 
     "🇮🇳",
     'title'),
    
    ("Jigsaw Rate Severity of Toxic Comments",
    "Rank relative ratings of toxicity between comments",
    "https://www.kaggle.com/c/jigsaw-toxic-severity-rating",
     "☣️",
     'total_votes'),
    ("TensorFlow - Help Protect the Great Barrier Reef",
    "Detect crown-of-thorns starfish in underwater image data",
    "https://www.kaggle.com/c/tensorflow-great-barrier-reef",
     "🐠",
     "total_votes"),
    
    
    ("Feedback Prize - Evaluating Student Writing",
    "Analyze argumentative writing elements from students grade 6-12",
    "https://www.kaggle.com/c/feedback-prize-2021",
     "📖",
     "total_votes"),
     
    
    ("H&M Personalized Fashion Recommendations",
    "Provide product recommendations based on previous purchases",
    "https://www.kaggle.com/c/h-and-m-personalized-fashion-recommendations",
     "H&M",
     "total_votes"),
    
    ("G-Research Crypto Forecasting",
    "Use your ML expertise to predict real crypto market data",
    "https://www.kaggle.com/c/g-research-crypto-forecasting",
    "💲" ,
     "total_votes"),
    
    ("BirdCLEF 2022",
    "Identify bird calls in soundscapes",
    "https://www.kaggle.com/c/birdclef-2022",
    "🐦",
     "total_votes"),
    
    ("Other competitions",
     "",
     "",
     is_other_competitions,
     "title")
         
]


def by_competition():
    txt = "# Organized by competition\n\n"
    for title, subtitle, url, matching_str, sort_by in COMPETITIONS:
        txt += get_section(df, title, subtitle, url, matching_str, sort_by)
        txt += "\n\n"
    return txt

In [174]:
txt_by_competition = by_competition()

print(txt_by_competition)

# Organized by competition

## [Sartorius - Cell Instance Segmentation](https://www.kaggle.com/c/sartorius-cell-instance-segmentation)
**Detect single neuronal cells in microscopy images**

* [🦠 Sartorius - Starter Torch Mask R-CNN [LB=0.273]](http://kaggle.com/julian3833/sartorius-starter-torch-mask-r-cnn-lb-0-273) (276)
* [🦠 Sartorius - Starter Baseline Torch U-net [>0.0]](http://kaggle.com/julian3833/sartorius-starter-baseline-torch-u-net-0-0) (173)
* [🦠 Sartorius - Classifier + Mask R-CNN [LB=0.28]](http://kaggle.com/julian3833/sartorius-classifier-mask-r-cnn-lb-0-28) (115)
* [🦠 Sartorius - Resnet34 Classifier](http://kaggle.com/julian3833/sartorius-resnet34-classifier) (43)
* [🦠 Sartorius - 3-line overlap-removing function](http://kaggle.com/julian3833/sartorius-3-line-overlap-removing-function) (13)

## [chaii - Hindi and Tamil Question Answering](https://www.kaggle.com/c/chaii-hindi-and-tamil-question-answering)
**Identify the answer to questions found in Indian language passage

# Organized by competition

## [Sartorius - Cell Instance Segmentation](https://www.kaggle.com/c/sartorius-cell-instance-segmentation)
**Detect single neuronal cells in microscopy images**

* [🦠 Sartorius - Starter Torch Mask R-CNN [LB=0.273]](http://kaggle.com/julian3833/sartorius-starter-torch-mask-r-cnn-lb-0-273) (277)
* [🦠 Sartorius - Starter Baseline Torch U-net [>0.0]](http://kaggle.com/julian3833/sartorius-starter-baseline-torch-u-net-0-0) (173)
* [🦠 Sartorius - Classifier + Mask R-CNN [LB=0.28]](http://kaggle.com/julian3833/sartorius-classifier-mask-r-cnn-lb-0-28) (115)
* [🦠 Sartorius - Resnet34 Classifier](http://kaggle.com/julian3833/sartorius-resnet34-classifier) (44)
* [🦠 Sartorius - 3-line overlap-removing function](http://kaggle.com/julian3833/sartorius-3-line-overlap-removing-function) (13)

## [chaii - Hindi and Tamil Question Answering](https://www.kaggle.com/c/chaii-hindi-and-tamil-question-answering)
**Identify the answer to questions found in Indian language passages**

* [1 - The competition [QA for QA noobs 🇮🇳]](http://kaggle.com/julian3833/1-the-competition-qa-for-qa-noobs) (37)
* [2 - The dataset [QA for QA noobs 🇮🇳]](http://kaggle.com/julian3833/2-the-dataset-qa-for-qa-noobs) (21)
* [3 - The metric (Jaccard) [QA for QA noobs 🇮🇳]](http://kaggle.com/julian3833/3-the-metric-jaccard-qa-for-qa-noobs) (20)
* [4 - Exploring Public Models [QA for QA noobs 🇮🇳]](http://kaggle.com/julian3833/4-exploring-public-models-qa-for-qa-noobs) (31)
* [5-🥇XLM-Roberta+Torch's extra data [LB:0.749] 🇮🇳](http://kaggle.com/julian3833/5-xlm-roberta-torch-s-extra-data-lb-0-749) (61)
* [6- 🤗 Pre & post-processing [QA for QA noobs 🇮🇳]](http://kaggle.com/julian3833/6-pre-post-processing-qa-for-qa-noobs) (30)
* [7 - Public Models Revisited [QA for QA noobs 🇮🇳]](http://kaggle.com/julian3833/7-public-models-revisited-qa-for-qa-noobs) (30)
* [Quick and Dirty Transliteration Tables 🇮🇳](http://kaggle.com/julian3833/quick-and-dirty-transliteration-tables) (13)

## [Jigsaw Rate Severity of Toxic Comments](https://www.kaggle.com/c/jigsaw-toxic-severity-rating)
**Rank relative ratings of toxicity between comments**

* [☣️ Jigsaw - Incredibly Simple Naive Bayes [0.768]](http://kaggle.com/julian3833/jigsaw-incredibly-simple-naive-bayes-0-768) (186)
* [☣️ Jigsaw - Early Ensemble [LB=0.836]](http://kaggle.com/julian3833/jigsaw-early-ensemble-lb-0-836) (104)
* [☣️ Jigsaw - 🤗 HF hub out-of-the-box models](http://kaggle.com/julian3833/jigsaw-hf-hub-out-of-the-box-models) (31)
* [☣️ Jigsaw - Explore Previous Competitions Datasets](http://kaggle.com/julian3833/jigsaw-explore-previous-competitions-datasets) (9)
* [☣️ Jigsaw - Random rank [LB=0.498]](http://kaggle.com/julian3833/jigsaw-random-rank-lb-0-498) (9)
* [☣️ Jigsaw - Jeremy Howard's NB [0.79744 Private]](http://kaggle.com/julian3833/jigsaw-jeremy-howard-s-nb-0-79744-private) (7)
* [☣️ Jigsaw - New Ensemble [LB=0.853]](http://kaggle.com/julian3833/jigsaw-new-ensemble-lb-0-853) (1)

## [TensorFlow - Help Protect the Great Barrier Reef](https://www.kaggle.com/c/tensorflow-great-barrier-reef)
**Detect crown-of-thorns starfish in underwater image data**

* [🐠 Reef - A CV strategy: subsequences!](http://kaggle.com/julian3833/reef-a-cv-strategy-subsequences) (221)
* [🐠 Reef- Starter Torch FasterRCNN Train [LB=0.416]](http://kaggle.com/julian3833/reef-starter-torch-fasterrcnn-train-lb-0-416) (137)
* [🐠 Reef- Starter Torch FasterRCNN Infer [LB=0.416]](http://kaggle.com/julian3833/reef-starter-torch-fasterrcnn-infer-lb-0-416) (91)
* [🐠 DETR - Detection Transformer - Train [0.189]](http://kaggle.com/julian3833/detr-detection-transformer-train-0-189) (32)
* [🐠 DETR - Detection Transformer - Infer [0.189]](http://kaggle.com/julian3833/detr-detection-transformer-infer-0-189) (27)
* [🐠 Reef - Minimal EDA](http://kaggle.com/julian3833/reef-minimal-eda) (4)

## [Feedback Prize - Evaluating Student Writing](https://www.kaggle.com/c/feedback-prize-2021)
**Analyze argumentative writing elements from students grade 6-12**

* [📖Feedback- Baseline🤗 Sentence Classifier [0.226]](http://kaggle.com/julian3833/feedback-baseline-sentence-classifier-0-226) (187)
* [📖 PyTorch- "ShortFormer" w/Chunks - Train [0.624]](http://kaggle.com/julian3833/pytorch-shortformer-w-chunks-train-0-624) (80)
* [📖 PyTorch- "ShortFormer" w/Chunks - Infer [0.624]](http://kaggle.com/julian3833/pytorch-shortformer-w-chunks-infer-0-624) (68)
* [📖 Topic Modeling with LDA](http://kaggle.com/julian3833/topic-modeling-with-lda) (15)
* [📖 Pytorch - ITPT - Intra-task pre-training](http://kaggle.com/julian3833/pytorch-itpt-intra-task-pre-training) (11)
* [📖 W&B-tracked Shortformers experiments](http://kaggle.com/julian3833/w-b-tracked-shortformers-experiments) (5)

## [H&M Personalized Fashion Recommendations](https://www.kaggle.com/c/h-and-m-personalized-fashion-recommendations)
**Provide product recommendations based on previous purchases**

* [H&M - Implicit ALS model [0.014]](http://kaggle.com/julian3833/h-m-implicit-als-model-0-014) (156)
* [H&M - Collaborative Filtering: User-user](http://kaggle.com/julian3833/h-m-collaborative-filtering-user-user) (60)
* [H&M - Content-based: 12 most popular items [0.007]](http://kaggle.com/julian3833/h-m-content-based-12-most-popular-items-0-007) (28)

## [G-Research Crypto Forecasting](https://www.kaggle.com/c/g-research-crypto-forecasting)
**Use your ML expertise to predict real crypto market data**

* [🪙💲 G-Research- Starter LGBM Pipeline](http://kaggle.com/julian3833/g-research-starter-lgbm-pipeline) (266)
* [🪙💲 G-Research- Using the overlap fully [LB=0.99]](http://kaggle.com/julian3833/g-research-using-the-overlap-fully-lb-0-99) (161)
* [🪙💲 Proposal for a meaningful LB + Strict LGBM](http://kaggle.com/julian3833/proposal-for-a-meaningful-lb-strict-lgbm) (121)
* [[S]🪙💲G-Research - Strict LGBM example [LB=0.017]](http://kaggle.com/julian3833/s-g-research-strict-lgbm-example-lb-0-017) (21)

## [BirdCLEF 2022](https://www.kaggle.com/c/birdclef-2022)
**Identify bird calls in soundscapes**

* [🐦 Audio 101. 2- Detailed EDA](http://kaggle.com/julian3833/audio-101-2-detailed-eda) (8)
* [🐦Audio 101. 1- Audio manipulation & musical notes](http://kaggle.com/julian3833/audio-101-1-audio-manipulation-musical-notes) (8)

## [Other competitions]()
****

* [1 - Quick start: read csv and flatten json fields](http://kaggle.com/julian3833/1-quick-start-read-csv-and-flatten-json-fields) (491)
* [1- Learning 🤗 - Out-of-the-box BERT [LB: 0.577]](http://kaggle.com/julian3833/1-learning-out-of-the-box-bert-lb-0-577) (11)
* [1- Learning 🤗 - Out-of-the-box BERT [LB: 0.8102]](http://kaggle.com/julian3833/1-learning-out-of-the-box-bert-lb-0-8102) (14)
* [2 - Quick study: LGBM, XGB and Catboost [LB: 1.66]](http://kaggle.com/julian3833/2-quick-study-lgbm-xgb-and-catboost-lb-1-66) (96)
* [2. Learning 🤗 - Out-of-the-box RoBERTa [LB: 0.53]](http://kaggle.com/julian3833/2-learning-out-of-the-box-roberta-lb-0-53) (18)
* [3- Learning 🤗 - Out-of-the-box Electra [LB: 0.58]](http://kaggle.com/julian3833/3-learning-out-of-the-box-electra-lb-0-58) (7)
* [GPT-2 Large –774M– w/Pytorch: Not that impressive](http://kaggle.com/julian3833/gpt-2-large-774m-w-pytorch-not-that-impressive) (14)
* [🚢 1 - Loading and visualizing the images](http://kaggle.com/julian3833/1-loading-and-visualizing-the-images) (34)
* [🚢 2 - Understanding & plotting rle bounding boxes](http://kaggle.com/julian3833/2-understanding-plotting-rle-bounding-boxes) (61)
* [🚢 3 - Basic exploratory analysis](http://kaggle.com/julian3833/3-basic-exploratory-analysis) (31)
* [🚢 4 - Exploring public models](http://kaggle.com/julian3833/4-exploring-public-models) (44)
* [🚢 5 - Submitting the test file (1.0 public LB)](http://kaggle.com/julian3833/5-submitting-the-test-file-1-0-public-lb) (35)